### DISCUSSION TITLE
Can't install FF3 + data importer on docker

### DISCUSSION THREAD

**Discussion by sevichenko** [firefly-iii/firefly-iii#8363] [General]
Hi! how are you? Happy new year!
[_firefly_iii_core_logs.txt](https://github.com/firefly-iii/firefly-iii/files/13865590/_firefly_iii_core_logs.txt)

i have tried several times and methods, but always looking on the installation guide.
When i install only FF3, i get it working at the first attempt.

The container firefly_iii_core is marked on portainer as unhealthy.

The log that i can download from there is this:

`[Mon Jan 08 19:29:48.320969 2024] [php:error] [pid 72] [client 127.0.0.1:59766] PHP Fatal error:  Uncaught Illuminate\\Encryption\\MissingAppKeyException: No application encryption key has been specified. in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php:79\nStack trace:\n#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/helpers.php(307): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}()\n#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(77): tap()\n#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(60): Illuminate\\Encryption\\EncryptionServiceProvider->key()\n#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php(32): Illuminate\\Encryption\\EncryptionServiceProvider->parseKey()\n#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Encryption\\EncryptionServiceProvider->Illuminate\\Encryption\\{closure}()\n#5 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()\n#6 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(957): Illuminate\\Container\\Container->resolve()\n#7 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()\n#8 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(942): Illuminate\\Container\\Container->make()\n#9 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1454): Illuminate\\Foundation\\Application->make()\n#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(213): Illuminate\\Container\\Container->offsetGet()\n#11 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(193): Illuminate\\Session\\SessionManager->buildEncryptedSession()\n#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(78): Illuminate\\Session\\SessionManager->buildSession()\n#13 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionManager.php(66): Illuminate\\Session\\SessionManager->createNativeDriver()\n#14 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php(106): Illuminate\\Session\\SessionManager->createFileDriver()\n#15 /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php(80): Illuminate\\Support\\Manager->createDriver()\n#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Session/SessionServiceProvider.php(52): Illuminate\\Support\\Manager->driver()\n#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(908): Illuminate\\Session\\SessionServiceProvider->Illuminate\\Session\\{closure}()\n#18 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(795): Illuminate\\Container\\Container->build()\n#19 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(957): Illuminate\\Container\\Container->resolve()\n#20 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(731): Illuminate\\Foundation\\Application->resolve()\n#21 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(942): Illuminate\\Container\\Container->make()\n#22 /var/www/html/vendor/laravel/framework/src/Illuminate/Container/Container.php(1454): Illuminate\\Foundation\\Application->make()\n#23 /var/www/html/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(130): Illuminate\\Container\\Container->offsetGet()\n#24 /var/www/html/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(96): Illuminate\\Auth\\AuthManager->createSessionDriver()\n#25 /var/www/html/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(70): Illuminate\\Auth\\AuthManager->resolve()\n#26 /var/www/html/vendor/laravel/framework/src/Illuminate/Auth/AuthManager.php(340): Illuminate\\Auth\\AuthManager->guard()\n#27 /var/www/html/app/Exceptions/Handler.php(198): Illuminate\\Auth\\AuthManager->__call()\n#28 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(730): FireflyIII\\Exceptions\\Handler->report()\n#29 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(650): Illuminate\\Foundation\\Exceptions\\Handler->renderHttpException()\n#30 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(556): Illuminate\\Foundation\\Exceptions\\Handler->prepareResponse()\n#31 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Exceptions/Handler.php(473): Illuminate\\Foundation\\Exceptions\\Handler->renderExceptionResponse()\n#32 /var/www/html/app/Exceptions/Handler.php(178): Illuminate\\Foundation\\Exceptions\\Handler->render()\n#33 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(218): FireflyIII\\Exceptions\\Handler->render()\n#34 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(195): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->renderHttpResponse()\n#35 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(231): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException()\n#36 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleShutdown()\n#37 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()\n#38 {main}\n  thrown in /var/www/html/vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php on line 79
`

What can i do?
thank you!

**Comment by JC5** [comment]
It looks like you haven't set any environment variables, especially APP_KEY.
